internal/api/http/error.go

```diff
@@ -11,37 +10,32 @@ func ZitadelErrorToHTTPStatusCode(err error) (statusCode int, ok bool) {
 	if err == nil {
 		return http.StatusOK, true
 	}
-	//nolint:errorlint
-	switch err.(type) {
-	case *zerrors.AlreadyExistsError:
+	switch {
+	case zerrors.IsErrorAlreadyExists(err):
 		return http.StatusConflict, true
-	case *zerrors.DeadlineExceededError:
+	case zerrors.IsDeadlineExceeded(err):
 		return http.StatusGatewayTimeout, true
-	case *zerrors.InternalError:
+	case zerrors.IsInternal(err):
 		return http.StatusInternalServerError, true
-	case *zerrors.InvalidArgumentError:
+	case zerrors.IsErrorInvalidArgument(err):
 		return http.StatusBadRequest, true
-	case *zerrors.NotFoundError:
+	case zerrors.IsNotFound(err):
 		return http.StatusNotFound, true
-	case *zerrors.PermissionDeniedError:
+	case zerrors.IsPermissionDenied(err):
 		return http.StatusForbidden, true
-	case *zerrors.PreconditionFailedError:
+	case zerrors.IsPreconditionFailed(err):
 		// use the same code as grpc-gateway:
 		// https://github.com/grpc-ecosystem/grpc-gateway/blob/9e33e38f15cb7d2f11096366e62ea391a3459ba9/runtime/errors.go#L59
 		return http.StatusBadRequest, true
-	case *zerrors.UnauthenticatedError:
+	case zerrors.IsUnauthenticated(err):
 		return http.StatusUnauthorized, true
-	case *zerrors.UnavailableError:
+	case zerrors.IsUnavailable(err):
 		return http.StatusServiceUnavailable, true
-	case *zerrors.UnimplementedError:
+	case zerrors.IsUnimplemented(err):
 		return http.StatusNotImplemented, true
-	case *zerrors.ResourceExhaustedError:
+	case zerrors.IsResourceExhausted(err):
 		return http.StatusTooManyRequests, true
 	default:
-		c := new(zerrors.ZitadelError)
-		if errors.As(err, &c) {
-			return ZitadelErrorToHTTPStatusCode(errors.Unwrap(err))
-		}
 		return http.StatusInternalServerError, false
```

Comment:
Well, actually this one isn't a point of PR, but I was working on errorslint fixes and found, that HTTP and gRPC MWs uses different approaches

IMHO, the [one from gRPC](https://github.com/zitadel/zitadel/blob/9ebf2316c671b21feef39d8c039440041969b916/internal/api/grpc/gerrors/zitadel_errors.go#L77) pkg is much more reliable and consistant in terms of errors detection, so here we are

Yet, there some questions about [this check](https://github.com/zitadel/zitadel/blob/9ebf2316c671b21feef39d8c039440041969b916/internal/api/grpc/gerrors/zitadel_errors.go#L73) and it's optimization since underlying `errors.As` calls in zerrors package